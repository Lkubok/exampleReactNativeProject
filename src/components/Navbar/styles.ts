import { StyleSheet } from 'react-native';

import { fontNames, variables } from 'res';

export default StyleSheet.create({
  inModal: {
    backgroundColor: variables.colors.black30,
    borderTopEndRadius: variables.borderRadius.small,
    borderTopStartRadius: variables.borderRadius.small,
  },
  navContainer: {
    borderBottomColor: variables.colors.inputBorder,
    borderBottomWidth: 0.5,
    flexDirection: 'row',
    minHeight: 54,
  },
  navContainerHasBottom: {
    height: 70,
    position: 'relative',
  },
  navContainerNoBorder: {
    borderBottomWidth: 0,
  },
  navContainerOverlay: {
    borderBottomColor: variables.colors.overlayHeaderBorder,
    borderBottomWidth: 1,
    minHeight: 54,
  },
  navContentBottom: {
    alignItems: 'center',
    bottom: 4,
    color: variables.colors.black,
    display: 'flex',
    fontFamily: fontNames.SFCompactTextSemibold,
    fontSize: variables.fontSize.base,
    letterSpacing: -0.41,
    position: 'absolute',
    width: '100%',
  },
  navContentCenter: {
    alignItems: 'center',
    flexGrow: 1,
    justifyContent: 'center',
    left: 0,
    minHeight: 54,
    position: 'absolute',
    right: 0,
    width: '100%',
    zIndex: 0,
  },
  navContentCenterHasBottom: {
    top: 8,
  },
  navContentCenterTitle: {
    color: variables.colors.black,
    fontFamily: fontNames.SFCompactTextSemibold,
    fontSize: variables.fontSize.base,
    letterSpacing: -0.41,
  },

  navContentLeft: {
    alignItems: 'flex-start',
    flexGrow: 1,
    justifyContent: 'center',
    paddingLeft: variables.spacing.medium,
    zIndex: 1,
  },
  navContentRight: {
    alignItems: 'center',
    flexDirection: 'row',
    flexGrow: 1,
    justifyContent: 'flex-end',
    paddingRight: variables.spacing.medium,
    zIndex: 1,
  },
});
