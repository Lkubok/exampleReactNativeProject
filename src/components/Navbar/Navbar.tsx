import React, { FunctionComponent, useCallback, useMemo } from 'react';
import { View, Text, TouchableOpacity, ViewStyle, StyleSheet, StyleProp } from 'react-native';
import { useNavigation } from '@react-navigation/native';

import { variables } from 'res/variables';
import ChevronLeft from 'assets/chevronLeft.svg';

import styles from './styles';

interface Props {
  bottom?: React.ReactNode;
  center?: React.ReactNode;
  left?: React.ReactNode;
  noBorder?: boolean;
  right?: React.ReactNode;
  withLinkBack?: boolean;
  style?: StyleProp<ViewStyle>;
  inModal?: boolean;
}

const Navbar: FunctionComponent<Props> = ({ bottom, center, left, noBorder, right, withLinkBack, inModal, style }) => {
  const { goBack } = useNavigation();

  const handleGoBack = useCallback(() => {
    withLinkBack && goBack();
  }, [goBack, withLinkBack]);

  const renderBackButton = useMemo(
    () => (
      <TouchableOpacity onPress={handleGoBack} hitSlop={{ bottom: 24, top: 24, left: 24, right: 24 }}>
        <ChevronLeft fill={variables.colors.accentRed} height={24} width={24} />
      </TouchableOpacity>
    ),
    [handleGoBack],
  );

  const renderLeft = useMemo(
    () => (
      <>
        {withLinkBack && renderBackButton}
        {left}
      </>
    ),
    [left, renderBackButton, withLinkBack],
  );

  const renderCenter = useMemo(
    () => (
      <View>
        <Text style={styles.navContentCenterTitle}>{center}</Text>
      </View>
    ),
    [center],
  );
  const renderRightSide = useMemo(() => <>{right}</>, [right]);
  const renderBottom = useMemo(() => <View style={styles.navContentBottom}>{bottom}</View>, [bottom]);

  return (
    <>
      <View
        style={StyleSheet.flatten([
          styles.navContainer,
          noBorder && styles.navContainerNoBorder,
          !!bottom && styles.navContainerHasBottom,
          inModal && styles.inModal,
          style && style,
        ])}>
        <View style={styles.navContentLeft}>{(withLinkBack || left) && renderLeft}</View>
        <View style={StyleSheet.flatten([styles.navContentCenter, !!bottom && styles.navContentCenterHasBottom])}>
          {center && renderCenter}
        </View>
        <View style={styles.navContentRight}>{right && renderRightSide}</View>
        {bottom && renderBottom}
      </View>
    </>
  );
};

export default Navbar;
