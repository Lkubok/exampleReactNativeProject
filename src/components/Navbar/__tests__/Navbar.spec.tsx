/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/unbound-method */
import React from 'react';
import { render } from 'utils';

import Navbar from '../Navbar';

jest.useFakeTimers();

describe('Home screen', () => {
  beforeEach(() => {});

  it('should match snapshot', () => {
    const { toJSON } = render(<Navbar />);
    expect(toJSON()).toMatchSnapshot();
  });
});
