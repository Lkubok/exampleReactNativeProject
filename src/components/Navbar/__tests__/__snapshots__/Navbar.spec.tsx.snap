// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "[7m[38;2;196;196;198m#c4c4c6[39m[27m",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "minHeight": 54,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexGrow": 1,
          "justifyContent": "center",
          "paddingLeft": 16,
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
          "left": 0,
          "minHeight": 54,
          "position": "absolute",
          "right": 0,
          "width": "100%",
          "zIndex": 0,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexGrow": 1,
          "justifyContent": "flex-end",
          "paddingRight": 16,
          "zIndex": 1,
        }
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
